### ISSUE TITLE
Blocks allow specifying argument types but do not enforce them

### ISSUE THREAD

**Issue by Yethal** [nushell/nushell#5286]
### Describe the bug

When specifying variables for a block it is possible to provide types for them but that type is not enforced. The only way for a block to throw a type-related error is to specify a nonexistent type

### How to reproduce

```
let block = {|x: bool| echo $x}
do $block "Definitely not a bool"
```

### Expected behavior

Either blocks do not allow types for variables and throw a syntax error or nushell enforces the type

### Configuration

|key|value|
|-|-|
|version|0.61.0|
|branch||
|tag||
|build_os|linux-x86_64|
|rust_version|rustc 1.59.0 (9d1b2106e 2022-02-23)|
|rust_channel|stable-x86_64-unknown-linux-gnu|
|cargo_version|cargo 1.59.0 (49d8809dc 2022-02-10)|
|pkg_version|0.61.0|
|build_time|2022-04-13 15:59:24 +02:00|
|build_rust_channel|release|
|features|dataframe, default, trash, which, zip|
|installed_plugins|gstat, inc, query, query json, query web, query xml|

**Comment by weirdan** [comment]
Looks like it has been implemented:

```
$ let block = {|x: bool| echo $x}; do $block "Definitely not a bool"
Error: nu::shell::cant_convert

  × Can't convert to bool.
   ╭─[entry #141:2:11]
 1 │ let block = {|x: bool| echo $x}
 2 │ do $block "Definitely not a bool"
   ·           ───────────┬───────────
   ·                      ╰── can't convert string to bool
   ╰────
```

**Comment by sholderbach** [comment]
Yep implemented in #12056
